### PR TITLE
Feat/BE-14-implement-Video-call-feature-Agora-token-generator

### PR DESCRIPTION
**Pull request by taiwonaf**
- Make the uid a querry parameter
- Removed the ability to generate token using uid

### CODE REVIEW

**Comment by billmal071** [comment]
@NwokoyeChigozie @Aminujb @UfiairENE Please help review and merge.

File: agora/rtctoken.go [hunk 1]
```diff
@@ -54,21 +53,20 @@ func generateRtcToken(appId, appCertificate, channelName, uidStr, tokentype stri
 			return "", err
 		}
 		return rtcToken, nil
+		// else if tokentype == "uid" {
+		// 	uid64, parseErr := strconv.ParseUint(uidStr, 10, 64)
+		// 	// check if conversion fails
+		// 	if parseErr != nil {
+		// 		err := fmt.Errorf("failed to parse uidStr: %s, to uint causing error: %s", uidStr, parseErr)
+		// 		return "", err
+		// 	}
```
Comment (Aminujb) [inline agora/rtctoken.go:62]: If these lines are not useful, you need to take them out.

File: agora/rtctoken.go [hunk 2]
```diff
@@ -54,21 +53,20 @@ func generateRtcToken(appId, appCertificate, channelName, uidStr, tokentype stri
...
-		if err != nil {
-			return "", err
-		}
-		return rtcToken, nil
+		// 	uid := uint32(uid64) // convert uid from uint64 to uint 32
+		// 	rtcToken, err := rtctokenbuilder.BuildTokenWithUID(appId, appCertificate, channelName, uid, role, expireTime)
+		// 	if err != nil {
+		// 		return "", err
+		// 	}nil
+		// 	return rtcToken,
```
Comment (Aminujb) [inline agora/rtctoken.go:69]: These too need to go if there is need for it.

**Review by Aminujb — CHANGES_REQUESTED** [review]
Just this minor change

**Review by Aminujb — APPROVED** [review]
LGTM!

**Pull request merged**